eslint.config.js

```diff
@@ -0,0 +1,97 @@
+import hypothesis from 'eslint-config-hypothesis';
+import jsxA11y from 'eslint-plugin-jsx-a11y';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+  {
+    ignores: [
+      '.tox/**/*',
+      '.yalc/**/*',
+      '.yarn/**/*',
+      'build/**/*',
+      '**/vendor/**/*.js',
+      '**/coverage/**/*',
+      'docs/_build/*',
+      'dev-server/static/**/*.js',
+    ],
+  },
+  ...hypothesis,
+  ...tseslint.configs.recommended,
+  jsxA11y.flatConfigs.recommended,
+  {
+    rules: {
+      'prefer-arrow-callback': [
+        'error',
+        {
+          allowNamedFunctions: true,
+        },
+      ],
+
+      'object-shorthand': ['error', 'properties'],
+      'react/prop-types': 'off',
+      '@typescript-eslint/no-unused-vars': 'error',
+      'no-use-before-define': 'off',
+
+      '@typescript-eslint/no-use-before-define': [
+        'error',
+        {
+          functions: false,
+          typedefs: false,
+          ignoreTypeReferences: false,
+        },
+      ],
+
+      '@typescript-eslint/ban-ts-comment': 'off',
+      '@typescript-eslint/no-empty-function': 'off',
+      '@typescript-eslint/no-explicit-any': 'off',
+      '@typescript-eslint/no-non-null-assertion': 'off',
+      '@typescript-eslint/no-this-alias': 'off',
+      '@typescript-eslint/consistent-type-assertions': 'error',
+      '@typescript-eslint/consistent-type-imports': 'error',
+
+      // Mocha
+      'mocha/no-mocha-arrows': 'off',
+      'mocha/no-setup-in-describe': 'off',
+      'mocha/max-top-level-suites': 'off',
+      'mocha/consistent-spacing-between-blocks': 'off',
+      'mocha/no-top-level-hooks': 'off',
+      'mocha/no-sibling-hooks': 'off',
+      'mocha/no-identical-title': 'off',
+    },
+  },
+
+  // Annotator module
+  {
+    files: ['src/annotator/**/*.{js|tx|tsx}'],
+    rules: {
+      'no-restricted-properties': [
+        2,
+        {
+          // Disable `bind` usage in annotator/ code to prevent unexpected behavior
+          // due to broken bind polyfills. See
+          // https://github.com/hypothesis/client/issues/245
+          property: 'bind',
+          message:
+            'Use function expressions instead of bind() in annotator/ code',
+        },
+      ],
+    },
+  },
+
+  // Scripts and configuration files
+  {
+    files: ['**/*.js'],
```

Comment:
I think I would prefer to use an allow-list rather than making these rules apply to all files and then attempting to exclude `src/`. It depends how cumbersome maintaining the allow-list is.

Reply:
Agree, but for now let's just migrate the config as-is (as much as possible). We can adjust that later.

Reply:
OK